Add GB National Insurance Numbers aka ID Numbers (#3032)
faker-ruby/faker: doc/default/id_number.md

# Generate a valid French Social Security number (INSEE number)
Faker::IdNumber.french_insee_number #=> "22510589696868"
```

## ID Number and Locales

> Besides the default ID numbers, faker supports localized `.valid` and `.invalid` values:

Reviewer: ```suggestion
## ID Number and Locales

Besides the default ID numbers, faker supports localized `.valid` and `.invalid` values:
```

# Generate a valid French Social Security number (INSEE number)
Faker::IdNumber.french_insee_number #=> "22510589696868"
```

## ID Number and Locales

> Besides the locale-specific ID Number methods. Faker supports retrieving localised calls to `.valid` and `.invalid`. 
> Here is an example:

Reviewer: ```suggestion
Besides the default ID Number methods, Faker supports localized `.valid` and `.invalid` values. Here is an example:
```

```ruby
Faker::Config.locale = 'fr-FR'
Faker::IdNumber.valid #=> "22510589696868"
```

Locales with specific intricacies are as such:

> ### en-GB
> When provided with British English, unformatted'[National Insurance](https://www.gov.uk/national-insurance/your-national-insurance-number)' numbers are generated. 
> Note: Faker can only generate a subset of all possible legal/illegal national insurance numbers.
> 
> ```ruby
> Faker::Config.locale = 'en-GB'
> Faker::IdNumber.valid #=> "AJ405924A"
> Faker::IdNumber.invalid #=> "BG316764W"
> ```

Reviewer: ```suggestion
### en-GB

When the locale is set to British English, unformatted'[National Insurance](https://www.gov.uk/national-insurance/your-national-insurance-number)' numbers are generated:

```ruby
Faker::Config.locale = 'en-GB'
Faker::IdNumber.valid #=> "AJ405924A"
Faker::IdNumber.invalid #=> "BG316764W"
```

Note: Faker generates a subset of all possible legal/illegal national insurance numbers.
